pkg/flist/flist.go

```diff
@@ -35,10 +41,19 @@ var (
 	ErrNotMountPoint                   = errors.New("path is not a mountpoint")
 	ErrTransportEndpointIsNotConencted = errors.New("transport endpoint is not connected")
 	ErrZFSProcessNotFound              = errors.New("0-fs process not found")
+	ErrHashNotSupported                = errors.New("hash not supported by flist host")
+	ErrHashInvalid                     = errors.New("invalid hash length")
 )
 
+// Hash type
+type Hash string
```

Comment:
do we need Hash and Path types? it seems they're only converted from a string and to string in the code

Reply:
The type alias just to make it clear when u reading the code and also give hints during compile time so u can't mix them by mistake. I did this because the method returns path and hash, if i use string `return (string, string, error)` it will never be clear which is which until u read docs or the code each time. But if i return (Hash, Path, error) it's clear and can't be mixed by accident, and only convert them back to string when needed

Reply:
You're right, I was thinking of changing the function signature to use named returns so would be `hash, path string, err error`